dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/outbound/NettyHttpRequestorWsoc10.java

```diff
@@ -315,17 +263,15 @@ public void close(VirtualConnection vc, Exception e) {
 
     @Override
     public WsByteBuffer completeResponse() throws IOException {
-        System.out.println("Completing response!");
         try {
-            // TODO Configure this with legacy timeouts
-            System.out.println("This should wait up to 60s");
-            responsePromise.get(60000, TimeUnit.MILLISECONDS);
+            responsePromise.get(HttpOption.READ_TIMEOUT.parse(httpOptions), TimeUnit.SECONDS);
         } catch (InterruptedException | ExecutionException | TimeoutException e1) {
-            System.out.println("How much did we wait?");
-            e1.printStackTrace();
+            // This most probably means we timed out waiting for a response and so
+            // we throw a socket timeout exception to wrap around the exception reached
+            throw new SocketTimeoutException(e1.getMessage());
         }
         if (resp == null) {
-            throw new IOException("Don't have a response yet! Do we need to wait for it?");
+            throw new IOException("Don't have a response yet!");
```

Comment:
I've added this PR to https://github.com/OpenLiberty/open-liberty/issues/31761 so we can revisit and see if we need translations for anything.